models/bert/__main__.py

```diff
@@ -87,19 +87,21 @@ def evaluate_split(model, processor, args, split='dev'):
     args.num_labels = dataset_map[args.dataset].NUM_CLASSES
     args.is_multilabel = dataset_map[args.dataset].IS_MULTILABEL
 
-    if args.do_train:
+    if not args.trained_model:
         save_path = os.path.join(args.save_path, dataset_map[args.dataset].NAME)
-        if os.path.exists(save_path) and os.listdir(save_path):
-            shutil.rmtree(save_path)
-        os.makedirs(save_path)
+        try:
+            os.makedirs(save_path)
+        except OSError as e:
+            if e.errno != errno.EEXIST:
+                raise
```

Comment:
It's better to explicitly raise a OSError here

Reply:
This was done for consistency, we don't do this for other models in the repo. Also, I remember having save_path_<timestamp>.pt for Castorini/Castor initially but it led to overflow of models. 
The safest way, to not run out of disk space, especially with BERT is to not have time-stamped paths.

Reply:
The error is raised if models/bert/saves/<dataset> exists due to race condition. We would want to ignore the unnecessary error if the path already exists due to previous run. Hence the change.

Reply:
It doesn't change the fact that incorrect eval and test metrics will be printed if two sessions are run simultaneously. Since the same bug exists for other models, they should also be fixed. I'll open an issue regarding this.

Reply:
I understand what lines 94 to 96 are for, and I don't have a problem with the addition. But it is better to explicitly raise an error such as OSError in line 96 in case e.errno != errno.EEXIST. Please make this change.